### DISCUSSION TITLE
Fullscreen video in a TikTok like flatlist

### DISCUSSION THREAD

**Discussion by zRelux** [expo/expo#17398] [Archived: Questions]
Has anyone here ever done a vertical video flatlist with a button that makes the video fullscreen (not the native UI). If yes could you give some advice on how to achieve it? Should I make the video position absolute and make it go from top to bottom (tried but couldn't make it work in the flatlist)? Should I have a hidden video component that shares the state with the one in the flatlist? Thanks any guidance is appreciated

**Comment by b0iq** [comment]
Hello mate,

Regarding you question my humble answer is:

You have to use like this:

Flatlist overlayed by buttons

```js
return(
  <>
  // Buttons likes comments...etc
  // Yes, it should have absolute position 
    <OverlayComponent />
  // Videos scroller
    <FlatList 
     snapToAlignment="start"
     decelerationRate={"fast"}
     snapToInterval={Dimensions.get("window").height}
     renderItem={videoRenderer}
     />
  </>
);
```

So, you have to delcare snapToInterval so you achieve full video scoll with snap.

If you need more help with that don't hesitate to let me know. 

Cheers,
